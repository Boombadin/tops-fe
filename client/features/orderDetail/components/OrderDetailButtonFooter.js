import React from 'react';
import styled from 'styled-components';

import withLocales from '@client/hoc/withLocales';
import { Loader } from '@client/magenta-ui';
import {
  ELEMENT_ACTION,
  ELEMENT_TYPE,
  generateTestId,
} from '@client/utils/generateElementId';

const ReorderButtonWrapper = styled.div`
  border-radius: 1px;
  background-color: #ffffff;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  padding: 0 28px;

  @media (max-width: 991px) {
    padding: 0 16px;
  }
`;

const Line = styled.div`
  width: 100%;
  height: 1px;
  border-top: solid 1px #ededed;
  margin-top: 9px;
`;

const FooterButtonWrapper = styled.div`
  width: 100%;
  display: flex;
  justify-content: flex-end;

  @media (max-width: 991px) {
    justify-content: center;
    flex-direction: column;
  }
`;

const TrackingButton = styled.button`
  width: 293px;
  height: 36px;
  border-radius: 5px;
  border: solid 1px #ed1d23;
  background-color: #ed1d23;
  color: #ffffff;
  font-size: 13px;
  font-weight: bold;
  margin: 31px 10px 29px 0;

  @media (max-width: 991px) {
    width: 100%;
    margin: 16px 0 6px 0;
  }
`;

const ReorderButton = styled.button`
  width: 293px;
  height: 36px;
  border-radius: 5px;
  border: solid 1px #ed1d23;
  background-color: #ffffff;
  color: #544943;
  font-size: 13px;
  font-weight: bold;
  margin: 31px 0 29px;

  @media (max-width: 991px) {
    width: 100%;
    margin: 0 0 22px 0;
  }
`;

const OrderDetailButtonFooter = ({
  orderId,
  translate,
  loadingReorder,
  orderTracking,
  handleTrackClick,
  handleReorderClick,
}) => {
  return (
    <ReorderButtonWrapper>
      <Line />
      <FooterButtonWrapper>
        {orderTracking && (
          <TrackingButton
            data-testid={generateTestId({
              type: ELEMENT_TYPE.BUTTON,
              action: ELEMENT_ACTION.VIEW,
              moduleName: 'OrderDetailButtonFooter',
              uniqueId: `handleTrackClick-${orderId}`,
            })}
            onClick={handleTrackClick}
            disabled={loadingReorder}
          >
            {translate('order_history.table.track_order')}
          </TrackingButton>
        )}
        <ReorderButton
          data-testid={generateTestId({
            type: ELEMENT_TYPE.BUTTON,
            action: ELEMENT_ACTION.ADD,
            moduleName: 'OrderDetailButtonFooter',
            uniqueId: `handleReorderClick-${orderId}`,
          })}
          onClick={handleReorderClick}
          disabled={loadingReorder}
        >
          {loadingReorder ? (
            <Loader
              data-testid={`view-loading-reorder-${orderId}`}
              active
              inline
              size="small"
            />
          ) : (
            translate('order_history.table.repeat')
          )}
        </ReorderButton>
      </FooterButtonWrapper>
    </ReorderButtonWrapper>
  );
};
export default withLocales(OrderDetailButtonFooter);
