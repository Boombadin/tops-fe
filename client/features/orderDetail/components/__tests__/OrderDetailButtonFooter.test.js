import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import OrderDetailButtonFooter from '../OrderDetailButtonFooter';

jest.mock('@client/hoc/withLocales', () => {
  return Component => props => (
    <Component translate={jest.fn(key => key)} {...props} />
  );
});

const TEST_ID = {
  BUTTON_TRACKORDER: 'btn-viewOrderDetailButtonFooter-handleTrackClick-809219',
  BUTTON_REORDER: 'btn-addOrderDetailButtonFooter-handleReorderClick-809219',
  LOADING: 'view-loading-reorder-809219',
};

describe('features/orderDetail/components/OrderDetailButtonFooter', () => {
  const props = {
    orderId: '809219',
    loadingReorder: false,
    orderTracking: true,
    handleTrackClick: () => {},
    handleReorderClick: () => {},
  };

  test('it should be match snapshot', () => {
    const { asFragment } = render(<OrderDetailButtonFooter {...props} />);
    expect(
      asFragment(<OrderDetailButtonFooter {...props} />),
    ).toMatchSnapshot();
  });

  test('it not show button tracking when tracking_info from api return array empty | it should be match snapshot', () => {
    const customProps = {
      orderId: '809219',
      loadingReorder: false,
      orderTracking: false,
      handleTrackClick: () => {},
      handleReorderClick: () => {},
    };

    const { asFragment } = render(<OrderDetailButtonFooter {...customProps} />);
    expect(
      asFragment(<OrderDetailButtonFooter {...customProps} />),
    ).toMatchSnapshot();
  });

  test('click open tracking process from grap | it should be match snapshot', () => {
    const { getByTestId, asFragment } = render(
      <OrderDetailButtonFooter {...props} />,
    );
    const buttonViewOrderTracking = getByTestId(TEST_ID.BUTTON_TRACKORDER);
    expect(buttonViewOrderTracking).toBeInTheDocument();
    fireEvent.click(buttonViewOrderTracking);
    expect(asFragment()).toMatchSnapshot();
  });

  test('click reorder | it should be match snapshot', () => {
    const { getByTestId, asFragment } = render(
      <OrderDetailButtonFooter {...props} />,
    );
    const buttonReorder = getByTestId(TEST_ID.BUTTON_REORDER);
    expect(buttonReorder).toBeInTheDocument();
    fireEvent.click(buttonReorder);
    expect(asFragment()).toMatchSnapshot();
  });

  test('click reorder when loading true | it should be match snapshot', () => {
    const propsUpdateLoading = {
      ...props,
      loadingReorder: true,
    };
    const { getByTestId, asFragment } = render(
      <OrderDetailButtonFooter {...propsUpdateLoading} />,
    );
    const loading = getByTestId(TEST_ID.LOADING);
    expect(loading).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
