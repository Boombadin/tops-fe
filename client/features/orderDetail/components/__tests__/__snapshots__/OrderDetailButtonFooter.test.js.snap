// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features/orderDetail/components/OrderDetailButtonFooter click open tracking process from grap | it should be match snapshot 1`] = `
<DocumentFragment>
  <div
    class="OrderDetailButtonFooter__ReorderButtonWrapper-sc-6rs6dw-0 fhILxp"
  >
    <div
      class="OrderDetailButtonFooter__Line-sc-6rs6dw-1 gtJCeu"
    />
    <div
      class="OrderDetailButtonFooter__FooterButtonWrapper-sc-6rs6dw-2 hSIxrd"
    >
      <button
        class="OrderDetailButtonFooter__TrackingButton-sc-6rs6dw-3 hhKNIt"
        data-testid="btn-viewOrderDetailButtonFooter-handleTrackClick-809219"
      >
        order_history.table.track_order
      </button>
      <button
        class="OrderDetailButtonFooter__ReorderButton-sc-6rs6dw-4 iHhFnX"
        data-testid="btn-addOrderDetailButtonFooter-handleReorderClick-809219"
      >
        order_history.table.repeat
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/orderDetail/components/OrderDetailButtonFooter click reorder | it should be match snapshot 1`] = `
<DocumentFragment>
  <div
    class="OrderDetailButtonFooter__ReorderButtonWrapper-sc-6rs6dw-0 fhILxp"
  >
    <div
      class="OrderDetailButtonFooter__Line-sc-6rs6dw-1 gtJCeu"
    />
    <div
      class="OrderDetailButtonFooter__FooterButtonWrapper-sc-6rs6dw-2 hSIxrd"
    >
      <button
        class="OrderDetailButtonFooter__TrackingButton-sc-6rs6dw-3 hhKNIt"
        data-testid="btn-viewOrderDetailButtonFooter-handleTrackClick-809219"
      >
        order_history.table.track_order
      </button>
      <button
        class="OrderDetailButtonFooter__ReorderButton-sc-6rs6dw-4 iHhFnX"
        data-testid="btn-addOrderDetailButtonFooter-handleReorderClick-809219"
      >
        order_history.table.repeat
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/orderDetail/components/OrderDetailButtonFooter click reorder when loading true | it should be match snapshot 1`] = `
<DocumentFragment>
  <div
    class="OrderDetailButtonFooter__ReorderButtonWrapper-sc-6rs6dw-0 fhILxp"
  >
    <div
      class="OrderDetailButtonFooter__Line-sc-6rs6dw-1 gtJCeu"
    />
    <div
      class="OrderDetailButtonFooter__FooterButtonWrapper-sc-6rs6dw-2 hSIxrd"
    >
      <button
        class="OrderDetailButtonFooter__TrackingButton-sc-6rs6dw-3 hhKNIt"
        data-testid="btn-viewOrderDetailButtonFooter-handleTrackClick-809219"
        disabled=""
      >
        order_history.table.track_order
      </button>
      <button
        class="OrderDetailButtonFooter__ReorderButton-sc-6rs6dw-4 iHhFnX"
        data-testid="btn-addOrderDetailButtonFooter-handleReorderClick-809219"
        disabled=""
      >
        <div
          class="ui small active inline loader"
          data-testid="view-loading-reorder-809219"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/orderDetail/components/OrderDetailButtonFooter it not show button tracking when tracking_info from api return array empty | it should be match snapshot 1`] = `
<DocumentFragment>
  <div
    class="OrderDetailButtonFooter__ReorderButtonWrapper-sc-6rs6dw-0 fhILxp"
  >
    <div
      class="OrderDetailButtonFooter__Line-sc-6rs6dw-1 gtJCeu"
    />
    <div
      class="OrderDetailButtonFooter__FooterButtonWrapper-sc-6rs6dw-2 hSIxrd"
    >
      <button
        class="OrderDetailButtonFooter__ReorderButton-sc-6rs6dw-4 iHhFnX"
        data-testid="btn-addOrderDetailButtonFooter-handleReorderClick-809219"
      >
        order_history.table.repeat
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`features/orderDetail/components/OrderDetailButtonFooter it should be match snapshot 1`] = `
<DocumentFragment>
  <div
    class="OrderDetailButtonFooter__ReorderButtonWrapper-sc-6rs6dw-0 fhILxp"
  >
    <div
      class="OrderDetailButtonFooter__Line-sc-6rs6dw-1 gtJCeu"
    />
    <div
      class="OrderDetailButtonFooter__FooterButtonWrapper-sc-6rs6dw-2 hSIxrd"
    >
      <button
        class="OrderDetailButtonFooter__TrackingButton-sc-6rs6dw-3 hhKNIt"
        data-testid="btn-viewOrderDetailButtonFooter-handleTrackClick-809219"
      >
        order_history.table.track_order
      </button>
      <button
        class="OrderDetailButtonFooter__ReorderButton-sc-6rs6dw-4 iHhFnX"
        data-testid="btn-addOrderDetailButtonFooter-handleReorderClick-809219"
      >
        order_history.table.repeat
      </button>
    </div>
  </div>
</DocumentFragment>
`;
